community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/InputEntityDecorators.java

```diff
@@ -75,15 +75,26 @@
         };
     }
 
-    public static <ENTITY extends InputEntity> Function<ENTITY,ENTITY> decorators(
-            final Function<ENTITY,ENTITY>... decorators )
+    public static <ENTITY extends InputEntity> Decorator<ENTITY> decorators(
+            final Decorator<ENTITY>... decorators )
     {
-        return from -> {
-            for ( Function<ENTITY,ENTITY> decorator : decorators )
+        return new Decorator<ENTITY>()
+        {
+            @Override
+            public ENTITY apply( ENTITY from )
+            {
+                for ( Decorator<ENTITY> decorator : decorators )
+                {
+                    from = decorator.apply( from );
+                }
+                return from;
+            }
+
+            @Override
+            public boolean isMutable()
             {
-                from = decorator.apply( from );
+                return Stream.of( decorators ).anyMatch( decorator -> decorator.isMutable() );
```

Comment:
can be replaced with method reference
`return Stream.of( decorators ).anyMatch( Decorator::isMutable );`